src/App.jsx

```diff
@@ -7,12 +10,86 @@ import SimulationResults from "./SimulationResults";
 function App() {
   const [builds, setBuilds] = useState([]);
 
+  const addBuild = () => {
+    setBuilds(builds.concat([{ id: uuid() }]));
+  };
+
+  const addRestriction = (buildId) => {
+    const build = _.find(builds, (build) => build.id === buildId);
+    if (_.isNil(build.restrictions)) {
+      build.restrictions = [];
+    }
+    build.restrictions.push({ id: uuid() });
+    setBuilds(_.cloneDeep(builds));
```

Comment:
If I don't use `_.cloneDeep`, it doesn't recognize a change. Seems weird. Is this right?

Reply:
The first rule of react is "don't mutate state"! So yes, you do need to deeply clone the state and set a new version, and I'd actually recommend you do the cloning first instead of waiting till the end. There are immutability libraries like immer that can help with this if it's a paradigm you have trouble with.